src/coreclr/scripts/superpmi_asmdiffs_checked_release.py

```diff
@@ -0,0 +1,205 @@
+                        "diff_jit_directory",
+                        lambda jit_directory: os.path.isdir(jit_directory),
+                        "diff_jit_directory doesn't exist")
+
+    coreclr_args.verify(args,
+                        "log_directory",
+                        lambda log_directory: True,
+                        "log_directory doesn't exist")
+
+    return coreclr_args
+
+
+def main(main_args):
+    """ Run superpmi asmdiffs process on the Helix machines.
+
+    See superpmi_asmdiffs_setup.py for how the directory structure is set up in the
+    correlation payload. This script lives in the root of that directory tree.
+
+    Args:
+        main_args ([type]): Arguments to the script
+    """
+
+    python_path = sys.executable
+    script_dir = os.path.abspath(os.path.dirname(os.path.realpath(__file__)))
+    coreclr_args = setup_args(main_args)
+
+    # It doesn't really matter where we put the downloaded SPMI artifacts.
+    # Here, they are put in <correlation_payload>/artifacts/spmi.
+    spmi_location = os.path.join(script_dir, "artifacts", "spmi")
+
+    log_directory = coreclr_args.log_directory
+    platform_name = coreclr_args.platform
+
+    # Find the built jit-analyze and put its directory on the PATH
+    jit_analyze_dir = os.path.join(script_dir, "jit-analyze")
+    if not os.path.isdir(jit_analyze_dir):
+        print("Error: jit-analyze not found in {} (continuing)".format(jit_analyze_dir))
+    else:
+        # Put the jit-analyze directory on the PATH so superpmi.py can find it.
+        print("Adding {} to PATH".format(jit_analyze_dir))
+        os.environ["PATH"] = jit_analyze_dir + os.pathsep + os.environ["PATH"]
+
+    # Find the portable `git` installation, and put `git.exe` on the PATH, for use by `jit-analyze`.
+    git_directory = os.path.join(script_dir, "git", "cmd")
+    git_exe_tool = os.path.join(git_directory, "git.exe")
+    if not os.path.isfile(git_exe_tool):
+        print("Error: `git` not found at {} (continuing)".format(git_exe_tool))
+    else:
+        # Put the git/cmd directory on the PATH so jit-analyze can find it.
+        print("Adding {} to PATH".format(git_directory))
+        os.environ["PATH"] = git_directory + os.pathsep + os.environ["PATH"]
+
+    # Figure out which JITs to use
+    os_name = "win" if platform_name.lower() == "windows" else "unix"
+    arch_name = coreclr_args.arch
+    host_arch_name = "x64" if arch_name.endswith("64") else "x86"
+    os_name = "universal" if arch_name.startswith("arm") else os_name
+    base_jit_path = os.path.join(coreclr_args.base_jit_directory, 'clrjit_{}_{}_{}.dll'.format(os_name, arch_name, host_arch_name))
+    diff_jit_path = os.path.join(coreclr_args.diff_jit_directory, 'clrjit_{}_{}_{}.dll'.format(os_name, arch_name, host_arch_name))
+
+    # Find out if it is asmdiffs between Checked binaries and Release binaries
+    diff_with_release = coreclr_args.diff_with_release
+    if diff_with_release is not True:
+        return 1
```

Comment:
Isn't this file only used for "diff_with_release" cases?

Or do you have hopes of commonizing this file and the very similar `superpmi_asmdiffs.py` file, to avoid code duplication?

Reply:
Removed.